gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/cockpit/command/handler/DeployModelCommandHandler.java

```diff
@@ -117,37 +132,59 @@ public Single<DeployModelReply> handle(DeployModelCommand command) {
-                }
-
-                result = cockpitApiService.createApi(
-                    executionContext,
-                    apiCrossId,
-                    user.getId(),
-                    swaggerDefinition,
-                    executionContext.getEnvironmentId(),
-                    mode,
-                    labels
-                );
-            }
-
-            if (result.isSuccess()) {
-                logger.info("Api imported [{}].", result.getApi().getId());
-                return Single.just(new DeployModelReply(command.getId()));
+                return message
+                    .map(error -> Single.just(new DeployModelReply(command.getId(), error)))
+                    .orElseGet(() -> {
+                        deployModelToApiCreateUseCase.execute(
+                            new DeployModelToApiCreateUseCase.Input(swaggerDefinition, audit, apiCrossId, fromDeploymentModel(mode), labels)
+                        );
+                        return Single.just(new DeployModelReply(command.getId()));
+                    });
             }
-            String errorDetails = "Failed to import API [%s].".formatted(result.getErrorMessage());
-            logger.error(errorDetails);
+        } catch (InvalidPathsException ipe) {
+            final var errorDetails = "Failed to import API [context path not available].";
+            logger.error(errorDetails, ipe);
             return Single.just(new DeployModelReply(command.getId(), errorDetails));
         } catch (Exception e) {
-            String errorDetails = "Error occurred when importing api [%s]".formatted(payload.modelId());
+            final var errorDetails = "Error occurred when importing api [%s]".formatted(payload.modelId());
             logger.error(errorDetails, e);
             return Single.just(new DeployModelReply(command.getId(), errorDetails));
         }
     }
 
+    private DeployModelToApiDomainService.Mode fromDeploymentModel(DeploymentMode mode) {
+        return switch (mode) {
+            case API_DOCUMENTED -> DeployModelToApiDomainService.Mode.DOCUMENTED;
+            case API_MOCKED -> DeployModelToApiDomainService.Mode.MOCKED;
+            case API_PUBLISHED -> DeployModelToApiDomainService.Mode.PUBLISHED;
+        };
+    }
+
+    private @NotNull Single<DeployModelReply> updateV2Api(
+        DeployModelCommand command,
+        String apiId,
+        ExecutionContext executionContext,
+        UserEntity user,
+        DeploymentMode mode,
+        String swaggerDefinition,
+        List<String> labels
+    ) {
+        var result = cockpitApiService.updateApi(
+            executionContext,
+            apiId,
+            user.getId(),
+            swaggerDefinition,
+            executionContext.getEnvironmentId(),
+            mode,
+            labels
+        );
+
+        logger.info("Api v2 updated [{}].", result.getApi().getId());
```

Comment:
Should we add the same log for v4 or remove that one (or putting in `debug`)
Maybe we should add it in the Usecases